ops/testing.py

```diff
@@ -2995,9 +2962,11 @@ def pull(self, path: str, *,
                 Union[BinaryIO, TextIO],
                 file_path.open("rb" if encoding is None else "r", encoding=encoding))
         except FileNotFoundError:
-            raise pebble.PathError('not-found', f'stat {path}: no such file or directory')
+            raise pebble.PathError('not-found',
+                                   f'stat {path}: no such file or directory') from None
```

Comment:
Are we sure we want to add `from None` to these?

Reply:
I think so, although I did either miss them or think otherwise when I was doing the last PR that added a bunch of "from" clauses to raising.

With no "from" at all, they all end up looking like:

```
Traceback (most recent call last):
...
ExceptionType: message

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
...
ExceptionType: message
```

This should really only be the case when there was an error during the exception handling. None of these are that - they are all converting the exception to a more meaningful type.

So that leaves the choice between "from None" and "from e". Do we want the charmer to know about the original exception, or is that just an implementation detail that could change in the future? To me these all seem like the latter.